app/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_status_next_transition_validator.py

```diff
@@ -0,0 +1,184 @@
+
+    @parameterized.expand(
+        [
+            [
+                NimbusExperiment.Status.DRAFT,
+                False,
+            ],
+            [
+                NimbusExperiment.Status.PREVIEW,
+                False,
+            ],
+            [
+                NimbusExperiment.Status.LIVE,
+                False,
+            ],
+            [
+                NimbusExperiment.Status.COMPLETE,
+                False,
+            ],
+        ]
+    )
+    def test_update_status_errors_for_status_next_draft(self, status, valid):
+        experiment = NimbusExperimentFactory.create(status=status)
+        serializer = NimbusExperimentSerializer(
+            experiment,
+            data={
+                "status_next": NimbusExperiment.Status.DRAFT,
+                "changelog_message": "test changelog message",
+            },
+            context={"user": self.user},
+        )
+        self.assertEquals(serializer.is_valid(), valid)
+
+    def test_allow_live_experiment_to_be_complete(self):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            lifecycle=NimbusExperimentFactory.Lifecycles.LAUNCH_APPROVE_APPROVE,
+            is_rollout=False,
+        )
+        serializer = NimbusExperimentSerializer(
+            experiment,
+            data={
+                "status": NimbusExperiment.Status.LIVE,
+                "publish_status": NimbusExperiment.PublishStatus.REVIEW,
+                "status_next": NimbusExperiment.Status.COMPLETE,
+                "changelog_message": "test changelog message",
+            },
+            context={"user": self.user},
+        )
+        self.assertTrue(serializer.is_valid())
+
+    def test_allow_live_experiment_to_be_dirty(self):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            lifecycle=NimbusExperimentFactory.Lifecycles.LAUNCH_APPROVE_APPROVE,
+            is_rollout=False,
+        )
+        serializer = NimbusExperimentSerializer(
+            experiment,
+            data={
+                "status": NimbusExperiment.Status.LIVE,
+                "publish_status": NimbusExperiment.PublishStatus.DIRTY,
+                "status_next": NimbusExperiment.Status.LIVE,
+                "changelog_message": "test changelog message",
+            },
+            context={"user": self.user},
+        )
+        self.assertTrue(serializer.is_valid())
+
+    def test_allow_live_rollout_to_be_dirty(self):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            lifecycle=NimbusExperimentFactory.Lifecycles.LAUNCH_APPROVE_APPROVE,
+            is_rollout=True,
+        )
+        serializer = NimbusExperimentSerializer(
+            experiment,
+            data={
+                "status": NimbusExperiment.Status.LIVE,
+                "publish_status": NimbusExperiment.PublishStatus.DIRTY,
+                "status_next": NimbusExperiment.Status.LIVE,
```

Comment:
here I understand the transition for live as status to live as status_next for the rollout